### PR TITLE
feature: moved l2-bridge-base to template-v2

### PR DESCRIPTION
**Pull request by sergeyWh1te**
* moved to template v2
* added unique findings
* added functional tests
* integrated healthCheck

### CODE REVIEW

File: l2-bridge-base/src/services/monitor_withdrawals.ts [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+
+        const finding: Finding = Finding.fromObject({
+          name: `⚠️ Base: Huge withdrawals during the last ` + `${Math.floor(period / (60 * 60))} hour(s)`,
+          description:
+            `There were withdrawals requests from L2 to L1 for the ` +
+            `${withdrawalsSum.div(ETH_DECIMALS).toFixed(4)} wstETH in total`,
+          alertId: 'HUGE-WITHDRAWALS-FROM-L2',
+          severity: FindingSeverity.Medium,
+          type: FindingType.Suspicious,
+          uniqueKey: getUniqueKey(uniqueKey, block.number),
```
Comment (TheDZhon) [inline l2-bridge-base/src/services/monitor_withdrawals.ts:117]: I'd suggest disambiguate blocks on L1 and L2 explicitly with the variable names, e.g.:
```ts
   uniqueKey: getUniqueKey(uniqueKey, l2Block.number),
```

File: l2-bridge-base/src/services/proxy_watcher.ts [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+          name: '🚨 Base: Proxy implementation changed',
+          description:
+            `Proxy implementation for ${this.proxyContract.getName()}(${this.proxyContract.getName()}) ` +
+            `was changed form ${this.getImpl()} to ${newImpl.right}` +
+            `\n(detected by func call)`,
+          alertId: 'PROXY-UPGRADED',
+          severity: FindingSeverity.High,
+          type: FindingType.Info,
+          metadata: { newImpl: newImpl.right, lastImpl: this.getImpl() },
+          uniqueKey: getUniqueKey(uniqueKey + '-' + this.proxyContract.getAddress(), blockNumber),
```
Comment (TheDZhon) [inline l2-bridge-base/src/services/proxy_watcher.ts:114]: Maybe it's better to pass an array of args to `getUniqueKey` to not pre-concatenate `uniqueKey` with address beforehand? :thinking:

File: l2-bridge-base/src/utils/events/bridge_events.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+      event:
+        'event RoleAdminChanged(bytes32 indexed role, bytes32 indexed previousAdminRole, bytes32 indexed newAdminRole)',
+      alertId: 'L2-BRIDGE-ROLE-ADMIN-CHANGED',
+      name: '🚨 Base L2 Bridge: Role Admin changed',
+      description: (args: Result) =>
+        `Role Admin for role ${args.role}(${RolesAddrToNameMap.get(args.role) || 'unknown'}) ` +
+        `was changed from ${args.previousAdminRole} to ${args.newAdminRole}`,
+      severity: FindingSeverity.High,
+      type: FindingType.Info,
+      uniqueKey: uniqueKeys[0],
```
Comment (TheDZhon) [inline l2-bridge-base/src/utils/events/bridge_events.ts:33]: sorry, but it's pretty tricky to follow the `uniqueKeys` indexes given that they presented out of order
maybe worth having a map instead to be more reader-friendly and maintainable?

**Review by TheDZhon — DISMISSED** [review]
LGTM

I suggest considering a few improvements :pray:

File: l2-bridge-base/src/services/monitor_withdrawals.ts (hunk 1 above)
Comment (sergeyWh1te) [reply l2-bridge-base/src/services/monitor_withdrawals.ts]: 59ea7c5676f9861e4d323cba22562145a73b14cd

File: l2-bridge-base/src/services/proxy_watcher.ts (hunk 2 above)
Comment (sergeyWh1te) [reply l2-bridge-base/src/services/proxy_watcher.ts]: 59ea7c5676f9861e4d323cba22562145a73b14cd

File: l2-bridge-base/src/utils/events/bridge_events.ts (hunk 3 above)
Comment (sergeyWh1te) [reply l2-bridge-base/src/utils/events/bridge_events.ts]: 59ea7c5676f9861e4d323cba22562145a73b14cd

**Review by TheDZhon — DISMISSED** [review]
Nice :+1:

File: l2-bridge-base/src/clients/base_provider.ts [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+
+export abstract class IProvider {
+  public abstract fetchL2Blocks(startBlock: number, endBlock: number): Promise<Block[]>
+
+  public abstract getL2Logs(startBlock: number, endBlock: number): Promise<E.Either<NetworkError, Log[]>>
+
+  public abstract getLatestL2Block(): Promise<E.Either<NetworkError, Block>>
+}
+
+export class BaseLineaProvider implements IProvider, IMonitorWithdrawalsClient {
```
[the remark this answers is not in the document]
Comment (sergeyWh1te) [reply l2-bridge-base/src/clients/base_provider.ts]: 99c048f9e7c2c4a734dd2433b47aa3be9d9a9e1e

File: l2-bridge-base/src/services/proxy_watcher.ts [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+
+        out.push(...impl, ...admin)
+      }
+    }
+
+    this.logger.info(elapsedTime(this.getName() + '.' + this.handleBlocks.name, start))
+    return out
+  }
+
+  private async handleProxyImplementationChanges(blockNumber: number): Promise<Finding[]> {
```
[the remark this answers is not in the document]
Comment (sergeyWh1te) [reply l2-bridge-base/src/services/proxy_watcher.ts]: 99c048f9e7c2c4a734dd2433b47aa3be9d9a9e1e

**Review by TheDZhon — APPROVED** [review]
:+1: Splendid

**Pull request merged**